### PR TITLE
Refactor dockerfile

### PR DESCRIPTION
**Pull request by tkdchen**

### CODE REVIEW

File: docker/install-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/bash
```
Comment (lcarva) [inline docker/install-dependencies.sh]: Let's add `set -euo pipefail` here. See http://redsymbol.net/articles/unofficial-bash-strict-mode/.

This way, we can run `dnf clean all` in this script as well.

Comment (tkdchen) [reply docker/install-dependencies.sh]: Done.

File: docker/install-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+set -euo pipefail
+
+# The URL to RCM Tools Fedora repos
+rcm_tools_repos=$1
```
Comment (lcarva) [inline docker/install-dependencies.sh]: I think you'll need to set a default value here. Empty string should do it.

Comment (tkdchen) [reply docker/install-dependencies.sh]: If build argument `rcm_tools_repo_file` is not set to `docker build`, `$1` will be expanded to an empty string eventually. This is an expected behavior. Sorry, I'm a little confused. I'm afraid it is unnecessary to set a default value here.

Comment (lcarva) [reply docker/install-dependencies.sh]: The `-u` flag will cause an error when accessing an undefined variable, in this case `$1`:
```
🐚 cat test.sh 
#!/bin/bash
set -euo pipefail
rcm_tools_repos=$1
echo 'It works!'
🐚 ./test.sh http://example.com/spam.repo
It works!
🐚 ./test.sh 
./test.sh: line 4: $1: unbound variable
```

You want this line to be `rcm_tools_repos=${1:-}`

Comment (tkdchen) [reply docker/install-dependencies.sh]: I get it. Thanks. :)

**Pull request merged**